### PR TITLE
add metric expression

### PR DESCRIPTION
**Pull request by jmazzitelli**

### CODE REVIEW

File: hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/inventory/MeasurementInstance.java [hunk 1]
```diff
@@ -75,4 +73,45 @@ public void setMetricLabels(Map<String, String> metricLabels) {
...
+                    metricString.append(comma)
+                            .append(label.getKey())
+                            .append("=\"")
+                            .append(label.getValue())
+                            .append("\"");
+                    comma = ",";
+                }
+                metricString.append("}");
+            }
+            expr = expr.replaceAll("\\$metric", metricString.toString());
```
Comment (jshaughn) [inline hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/inventory/MeasurementInstance.java]: I think you should use expr.replace() as opposed to replaceAll().  No need to deal with regex here.

Comment (jmazzitelli) [reply hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/inventory/MeasurementInstance.java]: I think thinking expressions can list it twice. Is there no way a prom query can specify the metric more than once in an expression?

Comment (jmazzitelli) [reply hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/inventory/MeasurementInstance.java]: fixed.. didn't realize replace does them all

**Comment by jshaughn** [comment]
closing in favor of https://github.com/hawkular/hawkular-agent/pull/416

**Pull request closed without merging**